mautic/mautic
# Add support for modal open/close callbacks.

…ess of the slow mQuery calls



| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ ]
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | [ ]
| Automated tests included?              | No 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 



#### Description:
This PR adds the ability to have custom callbacks on the show.bs.modal and hidden.bs.modal modal events. If you're using the data-toggle="ajaxmodal" method on your link in order to load a modal, you are now able to specify both a data-modal-open-callback and a data-modal-close-callback. The callback should be a string name of a method that exists on the Mautic global JS object. It will receive the HTML element that triggered the modal to open (so you can access its data attributes if needed.) If you wanted a callback to alert the user that the modal successfully opened, your link would look like this:

`<a href="whatever" data-toggle="ajaxmodal" data-target="#MauticSharedModal" data-modal-open-callback="alertUserItOpened">Open Modal</a>`
The callback method will look like this:

```
Mautic.alertUserItOpened = function(el) {
    alert('Hey it opened!');
};
```


#### Steps to test this PR:


1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Test that modals across the user interface can be opened and saved as they were before this change.
